File: DemoApp/DemoApp/MasterTableItem.m
//
//  MasterTableItem.m
//  DemoApp
//
//  Created by Glispa GmbH on 16.06.17.
//  Copyright © 2017 Glispa GmbH. All rights reserved.
//

#import "MasterTableItem.h"

@implementation MasterTableItem

- (id)initWithTitle:(NSString*)title withController:(NSString*)controller withXibName:(NSString*)xibName withChilds:(NSArray <MasterTableItem *>*)childItems withUserData:(NSDictionary <NSString *, NSObject *>*)userData withImageName:(NSString*)image {
    self = [super init];
    if (self) {
        self.title = title;
        self.controller = controller;
        self.xibName = xibName;
        self.childItems = childItems;
        self.userData = userData;
        self.image = image;
    }
    return self;
}

- (id)initWithTitle:(NSString*)title withChilds:(NSArray <MasterTableItem *>*)childItems withImageName:(NSString*)image {
    return [self initWithTitle:title withController:nil withXibName:nil withChilds:childItems withUserData:nil withImageName:image];
}

- (id)initWithTitle:(NSString*)title withController:(NSString*)controller withXibName:(NSString*)xibName withChilds:(NSArray <MasterTableItem *>*)childItems {
    return [self initWithTitle:title withController:controller withXibName:xibName withChilds:childItems withUserData:nil withImageName:nil];
}

- (id)initWithTitle:(NSString*)title withController:(NSString*)controller withXibName:(NSString*)xibName withChilds:(NSArray <MasterTableItem *>*)childItems withUserData:(NSDictionary <NSString *, NSObject *>*)userData {
    return [self initWithTitle:title withController:controller withXibName:xibName withChilds:childItems withUserData:userData withImageName:nil];
}

@end

File: DemoApp/DemoAppSwift/MasterTableItem.swift
//
//  MasterTableItem.swift
//  DemoAppSwift
//
//  Created by Glispa GmbH on 08.06.17.
//  Copyright ©  Glispa GmbH. All rights reserved.
//

import UIKit

struct MasterTableItem {
    let title: String
    let controller: String?
    let xibName: String?
    let childItems: [MasterTableItem]?
    let userData: [String:Any]?
    let image: String?
}

extension MasterTableItem {

    init(title: String, childItems: [MasterTableItem]?, image: String?) {
        self.init(title: title, controller: nil, xibName: nil, childItems: childItems, userData: nil, image: image)
    }

    init(title: String, controller: String?, xibName: String?, childItems: [MasterTableItem]?) {
        self.init(title: title, controller: controller, xibName: xibName, childItems: childItems, userData: nil, image: nil)
    }

    init(title: String, controller: String?, xibName: String?, childItems: [MasterTableItem]?, userData: [String:Any]?) {
        self.init(title: title, controller: controller, xibName: xibName, childItems: childItems, userData: userData, image: nil)
    }
}
